validphys2/src/validphys/covmats.py

```diff
@@ -690,12 +702,18 @@ def _covmat_t0_considered(covmat_t0_considered):
     return covmat_t0_considered
 
 
-def _dataset_inputs_covmat_t0_considered(dataset_inputs_covmat_t0_considered):
+def _dataset_inputs_covmat_t0_considered(dataset_inputs_covmat_t0_considered, fitthcovmat, data):
     """Helper function so we can dispatch the full
     covariance matrix accross dataset_inputs, having considered both ``use_t0``
     and ``use_pdferr``
     """
-    return dataset_inputs_covmat_t0_considered
+    if fitthcovmat is not None:
+        theory_covmat = fitthcovmat.load()
+        #change ordering according to exp_covmat (so according to runcard order)
+        tmp = theory_covmat.droplevel(0, axis=0).droplevel(0, axis=1)
+        bb = [str(i) for i in data]
+        return dataset_inputs_covmat_t0_considered + tmp.reindex(index=bb, columns=bb, level=0).values
```

Comment:
Could we use the new function in here as well? Also I have the impression one could encapsulate everything after `if fitthcovmat`